### PR TITLE
feat(HTTPError): set title to status message if title is None

### PR DESCRIPTION
**Pull request by ueg1990**
set HTTPError title attribute to status attribute if title is None

Fixes 778

### CODE REVIEW

File: falcon/http_error.py [hunk 1]
```diff
@@ -51,7 +52,7 @@ class HTTPError(Exception):
         status (str): HTTP status code and text, such as "400 Bad Request"
 
     Keyword Args:
-        title (str): Human-friendly error title (default ``None``).
+        title (str): Human-friendly error title.
```
Comment (kgriffs) [inline falcon/http_error.py]: We should explain that this arg is optional and if not passed, will default to the status.

**Comment by kgriffs** [comment]
Looks like codecov is complaining...

**Comment by ueg1990** [comment]
hey @kgriffs im not sure why the codecov is giving less than 100% even though I changed only one line of code...do you know what i am missing?

**Comment by kgriffs** [comment]
Looks like title can never be `None`, so this branch never executes: 

https://codecov.io/gh/falconry/falcon/src/17c18029de7d663d968df69bd71b51d4d2a367e6/falcon/http_error.py#L135

**Comment by ueg1990** [comment]
should i remove the line 'if self.title is not None' from lines 135 and 173?

**Comment by kgriffs** [comment]
That makes sense. Now that it can never be `None`, there's no need to check.

**Comment by kgriffs** [comment]
@ueg1990  Hmm, looks like at some point the history of this patch diverged from master. Would you mind rebasing just your commits onto master and doing a force push to clean up this branch (ueg1990:ueg1990_issue_778)?

**Comment by ueg1990** [comment]
How can I rebase only my commits? It looks like other commits are now included in this branch when i rebased before (Which i had not intend to do)

**Comment by kgriffs** [comment]
If all else fails you can create a new branch and then cherry-pick just the commits you want into that new branch.

**Comment by kgriffs** [comment]
Alternatively, you should be able to do an interactive rebase onto master (`git rebase -i master`) and then you can select which commits to keep.

**Comment by kgriffs** [comment]
[Rebased and merged as #809]

**Pull request closed without merging**